diarization_service.py

```
@@ -25,27 +27,55 @@ def __repr__(self):
+@contextlib.contextmanager
+def _as_wav(audio_path: str):
+    # MP3/FLAC headers report duration imprecisely; convert to WAV so pyannote
+    # gets an exact sample count without loading the whole file into RAM.
+    if audio_path.endswith(".wav"):
+        yield audio_path
+        return
+
+    fd, tmp_path = tempfile.mkstemp(suffix=".wav")
+    os.close(fd)
+    try:
+        subprocess.run(
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)